### ISSUE TITLE
Add PacIOOS models

### ISSUE THREAD

**Issue by kbailey-noaa** [asascience-open/IOOS-EDS#106]
From a recent meeting with PacIOOS, determined we need to add some PacIOOS models.  Will get a list from Jim Potemra/Melissa Iwamoto.

**Comment by kbailey-noaa** [comment]
This is a list I've discovered, but not asking for all to be ingested (yet) because I want to make sure they're registered in the IOOS catalog first:

Waves:  From http://www.pacioos.hawaii.edu/waves-category/model/:
Hawaii Regional Wave Model
Kauai Wave Model
Oahu Wave Model
Maui County Wave Model
Hawaii Island Wave Model
Northwestern Hawaiian Islands
Mariana Islands Regional Wave Model
Guam Island Regional Wave Model
Apra Harbor Regional Wave Model
Samoa Regional Wave Model
Tutuila/American Samoa Regional Wave Model

Currents:  From http://www.pacioos.hawaii.edu/currents-category/model/:
7 areas

Salinity:  From http://www.pacioos.hawaii.edu/water-category/model-salinity/
7 areas

Water Temperature:  http://www.pacioos.hawaii.edu/water-category/model-temperature/
7 areas

Winds:  http://www.pacioos.hawaii.edu/weather-category/model-wind/
- Except Global (I believe we already have this).

**Comment by Bobfrat** [comment]
@ericmbernier heads up

**Comment by kbailey-noaa** [comment]
I heard back from PacIOOS:  Currents, waves and winds are highest priority.  

The models are all registered in the IOOS Catalog, and I think they're here: https://data.ioos.us/dataset?tags=Models%2FAnalyses+%3E+%3E+Operational+Models&q=pacioos&res_format=OPeNDAP&_res_format_limit=0&_tags_limit=0

Before ingesting, please let me know the level of effort involved so I can make sure we're not burning through the budget on this.  If it's a lot we'll scale back.

**Comment by ericmbernier** [comment]
@kbailey-noaa Do you mind prioritizing the seven currents regions? Adding 40 or so new layers is a rather large task, thus I feel it would be best if we can rank these sources by area, and add a handful now, and the rest as time allows, in proper order.

**Comment by kbailey-noaa** [comment]
I agree -  Also, we don't need to add everything in the list this year.  I'd like to get a nice spread of RAs' sources, particularly output that is easy to add.  I'll reach out to PacIOOS and see if they have any priority/ranking of the 7 currents regions.  
I've been adding github issues for ingesting additional RA models, and I hope to have these all ranked/prioritized soon.  At this point it's helpful to know level of effort first.

**Comment by ericmbernier** [comment]
@kbailey-noaa I'd estimate a day per source (currents, winds, waves) for full EDS integration, which includes deploying to both our staging and production servers, and thorough testing on each. It could be less than a day each source, assuming everything goes seamlessly.

**Comment by kbailey-noaa** [comment]
@ericmbernier Ok, thanks.  I'm including @jpotemra on this, since he's the DMAC rep for PacIOOS.  He raised a good question >> ASA pulls output across on some regular interval to populate its own local database, as opposed to pointing to our TDS catalog.  If this is still the case, do you know the frequency with which they will pull the model output?  I imagine this would take a long while (some of the 4D model grids can be quite large). @kknee

**Comment by kknee** [comment]
@ericmbernier here's the priority list of PacIOOS layers from @kbailey-noaa - surface layer is fine to start with:

Mariana Islands Winds - WRF/ARW; 6-day, hourly forecast, 3 km resolution
Samoa Winds - WRF/ARW; 6-day, hourly forecast, 3 km resolution
Mariana Islands Regional Wave Model - WW3, 5-day hourly forecast, 5 km resolution
Samoa Regional Wave Model - WW3, 5-day hourly forecast, 5 km resolution
Hawaii Regional Wave Model - WW3, 5-day, hourly forecast, 5 km resolution
Apra Harbor Regional Wave Model - SWAN, 5-day, hourly forecast, 50 m resolution
Samoa Currents - ROMS, 7-day, 3-hourly forecast, 3 km resolution
Western North Pacific Currents - ROMS, 6-day, 3-hourly forecast, 4 km resolution
Mariana Islands Currents - ROMS, 6-day, 3-hourly forecast, 2 km resolution

@jpotemra - how frequently we pull the model output depends on how frequently the model is run (or how frequently you want it updated on EDS). Do you have any sources that are run more frequently than daily?

**Comment by jpotemra** [comment]
Hi Kelly,

All of our models are run daily (none more frequently).  My concern was 
that the download of all these files would take a while (e.g. the last 
one is a 7GB file).

Jim

**Comment by kknee** [comment]
@jpotemra - we're starting with the surface layers only, so the download should be substantially smaller than the original file size.  Also, we likely don't need all the variables, which will also help with download size.

**Comment by kknee** [comment]
@kbailey-noaa @jpotemra - update...we are now collecting Guam WRF, Samoa WRF, and Mariana WW3 with EDS.  Will get those on the development EDS instance ASAP.

**Comment by kknee** [comment]
Guam WRF, Samoa WRF, and Mariana WW3 can be viewed at http://dev.oceansmap.com/eds/#.  There is an issue with the Mariana legends that we're looking into.

**Comment by kknee** [comment]
Legend issue with Mariana WW3 has been resolved.  Please note - if your map view does not include the source AOI and you turn the layer on, the Table of Contents will note a warning for that layer

**Comment by kbailey-noaa** [comment]
@kknee Is the map view/AOI that gives a warning in the TOC going to be fixed too? I noticed that when I looked at dev.  If a user tries the layer and sees the error they aren't going to bother going to the AOI, so this could be a major issue.
The currents layers look good. The salinity layers load but the color ramp is still an issue, so it's just 1 or 2 colors. But can still move forward to production.  PacIOOS Marianas waves look good.
